src/PIL/Image.py

```diff
@@ -2240,13 +2240,20 @@ def thumbnail(self, size, resample=BICUBIC, reducing_gap=2.0):
         if x >= self.width and y >= self.height:
             return
 
+        def round_aspect(number):
+            if x / y >= aspect:
+                key = lambda n: abs(aspect - n / y)  # noqa: E731
+            else:
+                key = lambda n: abs(aspect - x / n)  # noqa: E731
```

Comment:
You can avoid this repeated condition if round_aspect receives a key as an argument:

``python
if x / y >= aspect:
    x = round_aspect(y * aspect, lambda n: abs(aspect - n / y))
else:
    y = round_aspect(x / aspect, lambda n: abs(aspect - x / n))
```

Reply:
That's a good suggestion, thanks!